File: JavaScript/OperatorASCIIArt.js
    function GetASCIIArt(character, size) {
      switch (character)
      {
        case "+":
          return "\r\n" + Plus(size);
        case "=":
            return "\r\n" + Equals(size);
        case "-":
          return "\r\n" + Minus(size);
        case "/":
          return "\r\n" + Divide(size);
        case "x":
          return "\r\n" + Multiply(size);
        default:
          return "";
      }
    }

    function Divide(size) {
      var output = "";

      for (d = size; d > 0; d--)
      {
        output += this.PadLeft("/", d) + "\r\n";
      }

      return output;
    }

    function Equals(size) {
      return Repeat("=", size) + "\r\n" + Repeat("=", size);
    }

    function Minus(size) {
      return Repeat("-", size);
    }

    function Multiply(size) {
      if (size == 1 || size == 2 || size == 3)
      {
        return "x";
      }

      var output = "";

      var correction = 0;

      var rows = new Array(Math.round(size));

      for (m = size; m > 0; m--)
      {
        if (size - correction - correction < 0)
        {
          output += this.PadLeft("x", correction +1) + "\r\n";
          break;
        }
        
        var row = this.Repeat(" ", correction)
          + "x"
          + this.Repeat(" ", size - correction - correction) + "x" + "\r\n";

        output += row;
        correction++;
        rows[correction] = row;

        if (m == size / 2)
        {
          break;
        }
      }

      for (m = rows.length - 1; m > 0; m--)
      {
        if (rows[m] == null || rows[m] == "")
        {
          continue;
        }
        output += rows[m];
      }

      return size % 2 == 0 ? "Multiply requires an odd number" : output;
    }

    function Plus( size) {
      var output = "";
	  
      for (p = 0; p < size; p++)
      {
        if (p == Math.round(size / 2) - 1)
        {
          output += Repeat("+", size) + "\r\n";
        }
        else
        {
          output += this.PadLeft("+", size - size / 2) + this.PadRight(" ", size - size / 2) + "\r\n";
        }
      }

      return size % 2 == 0 ? "Plus requires an odd number" : output;
    }

    function Repeat(repeatedCharacter, numberOfCharacters) {
        
        var output = "";
        
        for (r = 0; r < numberOfCharacters; r++) {
            output += repeatedCharacter;
        }
        
        return output;
    }
 
    function PadLeft(string, size){
        
		var chars = new Array(Math.round(size));
        
        for (i = 0; i < chars.length; i++) {
            chars[i] = " ";
        }
        
        chars[chars.length - 1] = string;
        
		var padLeftOutput = "";
		
		for (i = 0; i < chars.length - 1; i++) {
			padLeftOutput += chars[i];
		}
		
		return padLeftOutput + string;
    }
    
    function PadRight(string, size){
        
        var chars = new Array(Math.round(size));
        
        chars[0] = string;
        
        for (i = 1; i < chars.length; i++) {
            chars[i] = " ";
        }
        
		var padRightOutput = string;
		
		for (i = 0; i < chars.length - 1; i++) {
			padRightOutput += chars[i];
		}
		
        return padRightOutput;
    } 

File: Java/OperatorASCIIArt.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package scratchpad.CodeGolf;

/**
 *
 * @author Peter arden
 */
public class OperatorASCIIArt {
    
    public String GetASCIIArt(String character, int size) {
      switch (character)
      {
        case "+":
          return Plus(size);
        case "=":
            return Equals(size);
        case "-":
          return Minus(size);
        case "/":
          return Divide(size);
        case "x":
          return Multiply(size);
        default:
          return "";
      }
    }

    private String Divide(int size) {
      String output = "";

      for (int i = size; i > 0; i--)
      {
        output += this.PadLeft("/", i) + "\r\n";
      }

      return output;
    }

    private String Equals(int size) {
      return Repeat("=", size) + "\r\n" + Repeat("=", size);
    }

    private String Minus(int size) {
      return Repeat("-", size);
    }

    private String Multiply(int size) {
      if (size == 1 || size == 2 || size == 3)
      {
        return "x";
      }

      String output = "";

      int correction = 0;

      String[] rows = new String[size];

      for (int i = size; i > 0; i--)
      {
        if (size - correction - correction < 0)
        {
          //output += "x".PadLeft(correction + 1) + "\r\n";
          output += this.PadLeft("x", correction +1) + "\r\n";
          break;
        }
        
        String row = this.Repeat(" ", correction)
          + "x"
          + this.Repeat(" ", size - correction - correction) + "x" + "\r\n";

        output += row;
        correction++;
        rows[correction] = row;

        if (i == size / 2)
        {
          break;
        }
      }

      for (int i = rows.length - 1; i > 0; i--)
      {
        if (rows[i] == null || rows[i] == "")
        {
          continue;
        }
        output += rows[i];
      }

      return size % 2 == 0 ? "Multiply requires an odd number" : output;
    }

    private String Plus(int size) {
      String output = "";

      for (int i = 0; i < size; i++)
      {
        if (i == size / 2)
        {
          output += Repeat("+", size) + "\r\n";
        }
        else
        {
          output += this.PadLeft("+", size - size / 2) + this.PadRight(" ", size - size / 2) + "\r\n";
        }
      }

      return size % 2 == 0 ? "Plus requires an odd number" : output;
    }

    private String Repeat(String repeatedCharacter, int numberOfCharacters) {
        
        String output = "";
        
        for (int i = 0; i < numberOfCharacters; i++) {
            output += repeatedCharacter;
        }
        
        return output;
    }
 
    private String PadLeft(String string, int size){
        
        String[] chars = new String[size];
        
        for (int i = 0; i < chars.length - 1; i++) {
            chars[i] = " ";
        }
        
        chars[chars.length - 1] = string;
        
        return String.join("", chars);
    }
    
    private String PadRight(String string, int size){
        
        String[] chars = new String[size];
        
        chars[0] = string;
        
        for (int i = 1; i < chars.length; i++) {
            chars[i] = " ";
        }
        
        return String.join("", chars);
    }
}